pkg/middleware/auth_proxy/auth_proxy.go

```diff
@@ -274,6 +283,21 @@ func (auth *AuthProxy) LoginViaHeader() (int64, error) {
 	return upsert.Result.Id, nil
 }
 
+// headersIterator iterates over all non-empty supported additional headers
+func (auth *AuthProxy) headersIterator(fn func(field string, header string)) {
```

Comment:
I'm happy to hear an opinion on this approach. Is very common within [Ruby's enumerable](https://ruby-doc.org/core-2.6.3/Enumerable.html) library. Originally, I tried a function that converts the keys to a map and then other functions could just use it but this approach is a tiny bit more efficient (O(n) as opposed O(2n))